app/policies/plan_policy.rb

```diff
@@ -9,47 +9,47 @@ def initialize(user, plan)
   end
 
   def show?
-    @plan.readable_by?(@user.id)
+    @plan.readable_by?(@user.id) && Role.find_by(user_id: @user.id, plan_id: @plan.id).active
```

Comment:
It makes sense to keep this active check out of the `readable_by?` method. Is the check necessary though? These plans are filtered out of the user's dashboard, so the likelihood of them hitting these policy checks is slim. Even if they did via a bookmark or something, they should technically be able to access them.

This is maybe a discussion for a separate ticket though. I could see use cases where we would send them to these endpoints to view the plan and restore/unhide them if we decide to offer that option.